adapters/copy_test.go

```diff
@@ -0,0 +1,51 @@
+package adapters_test
+
+import (
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/adapters"
+	"github.com/smartcontractkit/chainlink/internal/cltest"
+	"github.com/stretchr/testify/assert"
+	"github.com/smartcontractkit/chainlink/store/models"
+)
+
+func TestCopy_Perform(t *testing.T) {
+	tests := []struct {
+		name            string
+		value           string
+		params          string
+		want            string
+		wantResultError bool
+	}{
+		{"existing path", `{"high":"11850.00","last":"11779.99"}`, `{"last": "value"}`,
+			`{"high":"11850.00","last":"11779.99","value":"11779.99"}`, false},
+		{"multiple copies", `{"high":"11850.00","last":"11779.99"}`, `{"last": "copiedLast", "high": "copiedHigh"}`,
+			`{"copiedHigh":"11850.00","copiedLast":"11779.99","high":"11850.00","last":"11779.99"}`, false},
+		{"float to key", `{"high":"11850.00","last":"11779.99"}`, `{"high": 1.1}`,
+			`{"1.1":"11850.00","high":"11850.00","last":"11779.99"}`, false},
+		{"nonexistent from key", `{"high":"11850.00","last":"11779.99"}`, `{"doesnotexist": "value"}`,
+			`{"high":"11850.00","last":"11779.99"}`, false},
+		{"empty to key", `{"high":"11850.00","last":"11779.99"}`, `{"high": ""}`,
+			`{"high":"11850.00","last":"11779.99"}`, false},
+		{"float value", `{"availability":0.99991}`, `{"availability": "value"}`,
+			`{"availability":0.99991,"value":"0.99991"}`, false},
```

Comment:
The inability to hit the error discussed above is highlighted here. We never have `wantResultError = true` in our test cases. I'm honestly ok with that because it is pretty hard to force that error, but let's still use `multierr` and remove `wantResultError` since it's never used.

Reply:
Let's add a test case to have coverage on the scenario where a destination is an object.

```
{"high": {"crazy": "object"}}
```

What do we want to happen here? The simple naive implementation is to just make a string like everything else:

```
{"{\"crazy\": \"object\"}}": "copiedvalue"}
```

We can revisit having deep copies or [having a `jq` adapter](https://github.com/savaki/jq) in a subsequent PR.